src/vivarium/framework/values.py

```diff
@@ -24,14 +25,41 @@
 from vivarium.manager import Interface, Manager
 from vivarium.types import NumberLike
 
+if TYPE_CHECKING:
+    from vivarium.framework.engine import Builder
+
+
+class ValueSource(Protocol):
```

Comment:
ooc, what was the mypy error that forced adding these interfaces?

Reply:
It might be possible to do this more compactly...We have to give the signatures of the Callables; For the value Source I you can make an alias like `Callable[... , NumberLike]` but I didn't know how to do a type alias for something like

Callable[[..., NumberLike, ...], NumberLike] for args and kwargs with a new positional argument in between. Maybe there is an answer in https://peps.python.org/pep-0612/#abstract